### ISSUE TITLE
Evaluation HMDB51

### ISSUE THREAD

**Issue by vateye** [kenshohara/3D-ResNets-PyTorch#40]
Hi Kenshohara, I'm not sure how to evaluate the HMDB51 test set on the model which is fine-tuned. And I wonder how to loaded HMDB51 pretrained weight on your model. I've tried but it report some dictionary errors.

**Comment by kenshohara** [comment]
Please tell me more details.
How did you try and what errors occur?

**Comment by vateye** [comment]
My input followed like this:
`python main.py --root_path ~/Research/datasets --video_path hmdb51/jpg --annotation_path hmdb51/hmdb51_2.json --result_path ~/Research/3D-ResNets-PyTorch/results/hmdb51 --dataset hmdb51 --n_classes 51 --model resnext --model_depth 101 --resnet_shortcut B --resnext_cardinality 32 --batch_size 32 --n_threads 4 --test --test_subset test --pretrain_path ~/Research/3D-ResNets-PyTorch/results/hmdb51/save_5.pth --no_train --no_val
`
Then I got 'test.log' file, but I'm not sure how to compute the average accuracy.

**Comment by kenshohara** [comment]
Please set —test-subset val.
I’m sorry for the confusing option.

**Comment by vateye** [comment]
In 'hmdb51_json.py'
`def convert_csv_to_dict(csv_dir_path, split_index):
    database = {}
    for filename in os.listdir(csv_dir_path):
        if 'split{}'.format(split_index) not in filename:
            continue
        
        data = pd.read_csv(os.path.join(csv_dir_path, filename),
                           delimiter=' ', header=None)
        keys = []
        subsets = []
        for i in range(data.shape[0]):
            row = data.ix[i, :]
            if row[1] == 0:
                _continue_
            elif row[1] == 1:
                subset = 'training'
            elif row[1] == 2:
                subset = 'validation'`
So, I can changed `continue` to 'subset = 'testing'. And evaluated it on the test set?

**Comment by vateye** [comment]
@kenshohara And meantime, I want to use train it on my own dataset. Is rewrite 'hmdb51_json.py' file to generate json file of my own dataset? (The dataset structure is same as generated n_frames. )

**Comment by kenshohara** [comment]
If you define the test set according to the code, you can evaluate on the test set.
The dictionary error may be caused by the different setting between the model of pretrain_path and the model specified by the options.

When you train a model on your own dataset,
you can use any format.
You should implement dataset class for your dataset,
and generate a dataset file for the dataset class.
